### PR TITLE
SUPESC-33: Adjusted the description of Braintree payment plugins and facade methods

### PR DESCRIPTION
**Pull request by dan-miniyarov-spryker**
## PR Description

Jira Ticket: https://spryker.atlassian.net/browse/SUPESC-33

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/technology-partner-guides/202204.0/payment-partners/braintree/installing-and-configuring-braintree.md [hunk 1]
```diff
@@ -195,11 +195,21 @@ All commands and conditions are located in `SprykerEco\Zed\Braintree\Communicati
 Default implementation for the checkout payment plugins should be added to `Pyz/Zed/Checkout/CheckoutDependencyProvider.php`.
 
 ```php
+protected function getCheckoutPreConditions(Container $container)
+{
+    $plugins = [
```
Comment (DmitryLymarenko) [inline docs/scos/dev/technology-partner-guides/202204.0/payment-partners/braintree/installing-and-configuring-braintree.md:200]: if this variable has no value, the please just update doc to return array

File: docs/scos/dev/technology-partner-guides/202108.0/payment-partners/braintree/braintree-performing-requests.md [hunk 2]
```diff
@@ -68,7 +68,8 @@ In order to perform the necessary requests in the project based on Spryker Comme
 | FACADE METHOD | PARAMETER | RETURN | DESCRIPTION |
 | --- | --- | --- | --- |
 |  `saveOrderPayment` | `QuoteTransfer``CheckoutResponseTransfer` | void | Saves order payment method data according to quote and checkout response transfer data. |
-|  `preCheckPayment` |  `QuoteTransfer` |  `BraintreeTransactionResponseTransfer` | Sends pre-authorize payment request to Braintree gateway to retrieve transaction data. Checks that form data matches transaction response data. |
+|  `preCheckPayment` |  `QuoteTransfer` |  `BraintreeTransactionResponseTransfer` | Sends pre-authorize payment request to Braintree gateway to retrieve transaction data. Checks that form data matches transaction response data. **This method is deprecated**: use `isQuotePaymentValid()` instead. |
+|  `isQuotePaymentValid` |  `QuoteTransfer` |  `BraintreeTransactionResponseTransfer` | Checks that quote has required data to perform a Braintree payment transaction. |
```
Comment (DmitryLymarenko) [inline docs/scos/dev/technology-partner-guides/202108.0/payment-partners/braintree/braintree-performing-requests.md:72]: don't we need to add new facade methods to this list?

Comment (dan-miniyarov-spryker) [reply docs/scos/dev/technology-partner-guides/202108.0/payment-partners/braintree/braintree-performing-requests.md:72]: There were 2 facade methods introduced, `isQuotePaymentValid()` and `executeCheckoutPostSaveHook`. They were both added

File: docs/scos/dev/technology-partner-guides/202204.0/payment-partners/braintree/installing-and-configuring-braintree.md (hunk 1 above)
Comment (dan-miniyarov-spryker) [reply docs/scos/dev/technology-partner-guides/202204.0/payment-partners/braintree/installing-and-configuring-braintree.md]: Fixed

**Review by DmitryLymarenko — DISMISSED** [review]

**Review by jeremyfourna — APPROVED** [review]
OK for me

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**